lib/ansible/modules/file.py

```diff
@@ -209,7 +209,16 @@
 
 '''
 RETURN = r'''
-
+dest:
+    description: Destination file/path, equal to the path argument
+    returned: state=touch, state=hard, state=link
+    type: str
+    sample: /path/to/file.txt
+path:
+    description: Destination file/path, equal to the path argument
```

Comment:
equal to dest, i always expect it to be equal to itself

Reply:
But it's not always equal to dest. See #55898 .

If you pass in `path=/tmp/file` with `state=touch` you get back `dest` and not `path`. If you pass in `dest=/tmp/dir` and `state=directory` you get back `path` and not `dest`.

With the module's existing functionality, whether or not `path` or `dest` is returned is completely unrelated to whether or not you pass in `path` or `dest`.

The documentation says `path` is the argument you pass in, and `dest` is an *alias*. So in the documentation I've chosen to only put in the main one. (Mentioning the alias throughout seems like it would add clutter.)

For the return value, it returns `path` or `dest` based solely on what the value of `state` is. 



Personally I'd like to change the behavior of the module to always return `path`, because consistency is intuitive. (And sometimes return `dest` for backwards compatability.) But I thought I'd start with just a documentation PR first.

Reply:
Clarified in f1a3251fe1303fb8d28536dbee7ac7382fad518d and 593d8231cd69e2e331a948efdbd0b26e8e4e3894.